arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedDictionaryEncodedParquetValuesReader.java

```diff
@@ -72,7 +72,7 @@ void readBatchOfDictionaryIds(IntVector intVector, int startOffset, int numValue
   }
 
   void readBatchOfDictionaryEncodedLongs(FieldVector vector, int startOffset, int numValuesToRead, Dictionary dict,
-                                         NullabilityHolder nullabilityHolder) {
+                                         NullabilityHolder nullabilityHolder, int typeWidth) {
```

Comment:
These changes look concerning. It looks like the old offset (only index) must not have been correct. If so, there are places where `getDataBuffer().setLong(...)` and similar methods are called but aren't updated like these. Are those cases bugs as well?

Reply:
Good catch, looks like I missed it. Will fix here and other places.

Reply:
Looks like the test `TestSparkParquetFallbackToDictionaryEncodingForVectorizedReader` wasn't adequately testing the fallback to plain encoding behavior for all the datatypes. Will tweak it.